static/app/views/replays/searchBar.tsx

```diff
@@ -0,0 +1,151 @@
+import {useEffect} from 'react';
+import assign from 'lodash/assign';
+import flatten from 'lodash/flatten';
+import memoize from 'lodash/memoize';
+import omit from 'lodash/omit';
+
+import {fetchTagValues} from 'sentry/actionCreators/tags';
+import SmartSearchBar from 'sentry/components/smartSearchBar';
+import {NEGATION_OPERATOR, SEARCH_WILDCARD} from 'sentry/constants';
+import {Organization, TagCollection} from 'sentry/types';
+import {defined} from 'sentry/utils';
+import {FIELD_TAGS, isAggregateField} from 'sentry/utils/discover/fields';
+import {FieldKind} from 'sentry/utils/fields';
+import useApi from 'sentry/utils/useApi';
+import withTags from 'sentry/utils/withTags';
+
+const SEARCH_SPECIAL_CHARS_REGEXP = new RegExp(
+  `^${NEGATION_OPERATOR}|\\${SEARCH_WILDCARD}`,
+  'g'
+);
+
+const REPLAY_TAGS = [
+  'browser.name',
+  'browser.version',
+  'countErrors',
+  'countSegments',
+  'device.brand',
+  'device.family',
+  'device.model',
+  'device.name',
+  'dist',
+  'duration',
+  'id',
+  'os.build',
+  'os.kernel_version',
+  'platform.name',
+  'projectId',
+  'release',
+  'sdk.name',
+  'sdk.version',
+  'user.display',
+  'user.email',
+  'user.id',
+  'user.ip',
+  'user.username',
+];
+
+const getReplayTags = () =>
+  Object.fromEntries(
+    REPLAY_TAGS.map(key => [
+      key,
+      {
+        ...FIELD_TAGS[key],
+        kind: FieldKind.FIELD,
+      },
+    ])
+  );
+
+export type SearchBarProps = Omit<React.ComponentProps<typeof SmartSearchBar>, 'tags'> & {
+  organization: Organization;
+  tags: TagCollection;
+  /**
+   * Used to define the max height of the menu in px.
+   */
+  maxMenuHeight?: number;
+  maxSearchItems?: React.ComponentProps<typeof SmartSearchBar>['maxSearchItems'];
+  omitTags?: string[];
+  projectIds?: number[] | Readonly<number[]>;
+};
+
+function SearchBar(props: SearchBarProps) {
+  const {maxSearchItems, organization, tags, omitTags, projectIds, maxMenuHeight} = props;
+
+  const api = useApi();
+
+  useEffect(() => {
+    // Clear memoized data on mount to make tests more consistent.
+    getEventFieldValues.cache.clear?.();
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [projectIds]);
+
+  // Returns array of tag values that substring match `query`; invokes `callback`
+  // with data when ready
+  const getEventFieldValues = memoize(
+    (tag, query, endpointParams): Promise<string[]> => {
+      const projectIdStrings = (projectIds as Readonly<number>[])?.map(String);
```

Comment:
The types here don't make sense to me. If we're forcing it to be a `Readonly<number>[]` then we don't need the optional-chaining `?` in there.

Shouldn't need to add anything extra imo:
```suggestion
      const projectIdStrings = projectIds?.map(String);
```